spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkFilterUtils.java

```diff
@@ -338,6 +338,8 @@ static String escape(String value) {
   }
 
   static String quote(String value) {
-    return "`" + value + "`";
+    return value.contains(".")
+        ? Arrays.stream(value.split("\\.")).collect(Collectors.joining("`.`", "`", "`"))
```

Comment:
could we add a comment to describe better what's going on here?

Reply:
Added comment, thanks for the suggestion